### PR TITLE
test_relationships_tables.py metrics pod fix

### PR DESCRIPTION
**Pull request by jdemon519**
**Purpose or Intent
=================**

Fixed test_relationships_tables.py to account for two pods lacking replicators associated with them. Also made sure that comments adhere to the proper format.

Added test_return_links.py

{{pytest: cfme/tests/containers/test_relationships_tables.py -v --use-provider container }}

### CODE REVIEW

**Comment by jdemon519** [comment]
Guys, this is the revised

https://github.com/ManageIQ/integration_tests/pull/3291 

Should I squash it down some more or this is ok?

**Comment by dajoRH** [comment]
Lint report for commit 4299904ac42abe1613ee5fce0d5f60b990d4cc8f:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 8056566c12a04e3c953b3d59737cf5ba4375e1d4:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit dd2a0e8e35480d81e44d4d59561745aaee9e1d78:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by jdemon519** [comment]
Jenkins test run

https://cm-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/view/All/job/ashlifst-test/61/console

**Comment by dajoRH** [comment]
Lint report for commit 74d1d429a0452236a2e78dedf1b7f2ec16b3d5e4:

Everything seems all right :smile: .
*CFME QE Bot*

**Comment by psav** [comment]
As mentioned before, you can't have multiple {{pytest: xxx}} statements in the GH description, only one you will need to modify the logic of the pytest command to select the tests you care about.

**Comment by jdemon519** [comment]
@psav Right, I have edited the GH statement

File: cfme/tests/containers/test_imgreg_proj_views.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    assert tb.is_active(
+        'List View'), "Image Registries list view setting failed"
+
+# CMP-9886
+
+
+def test_projects_list_views():
+    sel.force_navigate('containers_projects')
+    assert tb.is_active(
+        'List View'), "Projects list view setting failed"
```
Comment (psav) [inline cfme/tests/containers/test_imgreg_proj_views.py]: Not sure I understand this test, as it looks like there was nothing set to begin with. In this was you can't assert behaviour if 
a) you didn't specify what it _should_ be or
b) if could have been changed elsewhere by another test

File: cfme/tests/containers/test_one_image_analysis.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+pytestmark = [
+    pytest.mark.uncollectif(
+        lambda: current_version() < "5.6"),
+    pytest.mark.usefixtures('setup_provider'),
+    pytest.mark.tier(1)]
+pytest_generate_tests = testgen.generate(
+    testgen.container_providers, scope="function")
+
+
+def test_image_analysis_finished():
```
Comment (psav) [inline cfme/tests/containers/test_one_image_analysis.py]: This is addressed in #3330

File: cfme/tests/containers/test_pods_prop_data_integrity.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    """
+    sel.force_navigate('containers_pods')
+    list_tbl_pod = CheckboxTable(table_locator="//div[@id='list_grid']//table")
+    ui_pods = [r.name.text for r in list_tbl_pod.rows()]
+    ui_pods_revised = filter(
+        lambda ch: not ch.startswith('metrics'),
+        ui_pods)
+
+    for name in ui_pods_revised:
+        obj = Pod(name, provider)
```
Comment (psav) [inline cfme/tests/containers/test_pods_prop_data_integrity.py]: There is too much UI interaction going on here, this test should be something more like
```python
for pod in Provider.get_all_pods():
        # Perhaps need to check it's not the "metric" one in here, or make the get_all_pods take some kind of parameter to allow you to specificy
        val_cont_stat_summary = obj.get_detail(
            'Container Statuses Summary', 'Running')
        val_cont_rel_tbl = obj.get_detail('Relationships', 'Containers')
        assert val_cont_rel_tbl == val_cont_stat_summary
```

It's likely that ```get_all_pods``` as a function doesn't exist, but it should be done this way. It should also return Pod objects, not string names etc.

File: cfme/tests/containers/test_relationships_tables.py [hunk 4]
```diff
@@ -33,17 +31,21 @@
...
+    """ This module verifies the integrity of the Relationships table
+          We also verify that clicking on the Relationships table field
+          takes the user to the correct page, and the number of rows
+          that appears on that page is equal to the number in the
+          Relationships table
+    """
+
     sel.force_navigate('containers_pods')
     list_tbl_pod = CheckboxTable(table_locator="//div[@id='list_grid']//table")
     ui_pods = [r.name.text for r in list_tbl_pod.rows()]
```
Comment (psav) [inline cfme/tests/containers/test_relationships_tables.py]: Same as above

File: cfme/tests/containers/test_return_links.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    ui_services = [r.name.text for r in list_tbl_service.rows()]
+
+    for name in ui_services:
+        obj = Service(name, provider)
+        obj.navigate()
+        sel.click(service.link_to_prv_page)
+        assert "Services" in sel.title()
+
+
+def test_routes_link(provider):
```
Comment (psav) [inline cfme/tests/containers/test_return_links.py]: Same as above, these tests shouldn't be doing so much UI interaction inside the test

File: cfme/tests/containers/test_one_image_analysis.py (hunk 2 above)
Comment (jdemon519) [inline cfme/tests/containers/test_one_image_analysis.py]: Fixed in #3330

File: cfme/tests/containers/test_pods_prop_data_integrity.py (hunk 3 above)
Comment (jdemon519) [inline cfme/tests/containers/test_pods_prop_data_integrity.py]: I have modified it by using obj.summary, it verifies the number of pods that is displayed in one table is equal to the number displayed in the other (Relationships_ table). We don't have anything resembling get_all_pods function, but if we were to return the number of pods, what would be the best way to go about it? Through mgmt?

**Comment by dajoRH** [comment]
Lint report for commit 8c337a57bbfc3a58f3edaee5b377fcb76804e231:

`cfme/tests/containers/test_pods_prop_data_integrity.py`:
- :heavy_exclamation_mark: Line 40:1: **W391** *blank line at end of file*

Please, rectify these issues :smirk: .
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 1b2fd46ce0667e96d505990d5259bb398c8a0b43:

Everything seems all right :smile: .
*CFME QE Bot*

File: cfme/tests/containers/test_pods_prop_data_integrity.py (hunk 3 above)
Comment (psav) [inline cfme/tests/containers/test_pods_prop_data_integrity.py]: No, remember mgmt will bypass cfme, this is not what you want to assert that CFME is showing the right information. So I think most of what you have here is probably OK, but you want to move it inside the Pod module. it should be a classmethod (meaning it doesn't need and instance of a Pod to work)

**Comment by jkrocil** [comment]
Incomplete / incorrect description, as with https://github.com/ManageIQ/integration_tests/pull/3330.

**Pull request closed without merging**